<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>results module &mdash; PLACEHOLDER 1.0.0 documentation</title><link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="software module" href="software.html" />
    <link rel="prev" title="quality module" href="quality.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            PLACEHOLDER
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">src</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="alignment.html">alignment module</a></li>
<li class="toctree-l2"><a class="reference internal" href="arguments.html">arguments module</a></li>
<li class="toctree-l2"><a class="reference internal" href="calling.html">calling module</a></li>
<li class="toctree-l2"><a class="reference internal" href="config.html">config module</a></li>
<li class="toctree-l2"><a class="reference internal" href="cropping.html">cropping module</a></li>
<li class="toctree-l2"><a class="reference internal" href="downloads.html">downloads module</a></li>
<li class="toctree-l2"><a class="reference internal" href="logger.html">logger module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline.html">pipeline module</a></li>
<li class="toctree-l2"><a class="reference internal" href="quality.html">quality module</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">results module</a></li>
<li class="toctree-l2"><a class="reference internal" href="software.html">software module</a></li>
<li class="toctree-l2"><a class="reference internal" href="util.html">util module</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PLACEHOLDER</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">src</a></li>
      <li class="breadcrumb-item active">results module</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/results.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="module-results">
<span id="results-module"></span><h1>results module<a class="headerlink" href="#module-results" title="Permalink to this headline">¶</a></h1>
<p>This module handles everything about the final analysis and results 
output.</p>
<dl class="py function">
<dt id="results.mergeCalling">
<code class="sig-prename descclassname">results.</code><code class="sig-name descname">mergeCalling</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em><span class="sig-paren">)</span><a class="headerlink" href="#results.mergeCalling" title="Permalink to this definition">¶</a></dt>
<dd><p>Merges the results from REDItools2, JACUSA, SnpEff and 
bcftools mpileup, and generates the final Excel files and the 
graphs.</p>
<p>It reads the REDItools2 VCF and merges it with the SnpEff
REDItools2 VCF file by the SNV position, so the SNVs found by 
REDItools2 that were not found by SnpEff are discarded. Then, it 
filters the SNVs by the minimum SNV coverage and the minimum read 
support and saves those results in a different dataframe. These
are saved into an Excel file named reditools.xlsx, and it produces 
a count bar plot named reditools.count.png.</p>
<p>Then, it reads the JACUSA VCF and merges it with the SnpEff
JACUSA VCF file by the SNV position, so the SNVs found by 
JACUSA that were not found by SnpEff are discarded. Then, it 
filters the SNVs by the minimum SNV coverage and the minimum read 
support and saves those results in a different dataframe. These
are saved into an Excel file named jacusa.xlsx, and it produces 
a count bar plot named jacusa.count.png.</p>
<p>Finally, both filtered dataframes are merged by the SNV position,
so that only the SNVs common to both outputs remain. These
are saved into an Excel file named runCommon.xlsx, and it produces 
a count bar plot named common.count.png.</p>
<p>The original dataframes that were not filtered by minimum SNV
coverage and minimum read support are merged, so that only the 
SNVs common to both outputs remain. These are saved into an Excel 
file named runUnfilteredCommon.xlsx, and it produces a count bar 
plot named unfiltered.count.png.</p>
<p>The JACUSA, REDItools2, common and unfiltered dataframes are 
concatenated by dataframe type to the dataframes of the other
runs. In the end, there will be a global JACUSA dataframe, a 
global REDItools2 dataframe, a global common dataframe and a 
global unfiltered dataframe.</p>
<p>This global unfiltered dataframe is processed further, so that only
the SNVs that appear in multiple runs remain, and they are merged
into a single row per SNV. These are saved into an Excel file named
globalRecurring.xlsx, and it produces a count bar plot named
recurring.count.png.</p>
<p>The JACUSA dataframe is saved into an Excel file named 
globalJacusa.xlsx and it produces a count bar plot named 
jacusa.count.png, but at the genome level. Same with REDItools2
(globalReditools.xlsx) and common (globalCommon.xlsx) dataframes.</p>
<p>Finally, it produces a strip plot with the frequencies of the SNVs 
and a histogram with the position of the SNVs per mutation with the
data from the global common dataframe.</p>
<p>All this process is repeated for every reference viral genome. The
output files are found in the 6-visualization directory.</p>
<dl class="simple">
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><p>0 - A list of the paths for the input run (one file if single-end, two if paired-end)
1 - Run type. “single” if single-end, “paired” if paired-end
2 - Run ID</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="results.runSnpEff">
<code class="sig-prename descclassname">results.</code><code class="sig-name descname">runSnpEff</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em><span class="sig-paren">)</span><a class="headerlink" href="#results.runSnpEff" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs SnpEff on the VCF files resulting from JACUSA and REDItools2.</p>
<p>First, it creates a configuration SnpEff file, adding the database
names for every reference genome. Then, it creates those databases.
Finally, each VCF file is analyzed against each database and outputs
a VCF file per run per tool. These files’ names end with
.snpeff.jacusa.vcf and .snpeff.reditools.vcf, and are found in the 
5-snpeff directory.</p>
<dl class="simple">
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="quality.html" class="btn btn-neutral float-left" title="quality module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="software.html" class="btn btn-neutral float-right" title="software module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, David Guevara.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>