<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>alignment module &mdash; PLACEHOLDER 1.0.0 documentation</title><link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="arguments module" href="arguments.html" />
    <link rel="prev" title="src" href="modules.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            PLACEHOLDER
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">src</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">alignment module</a></li>
<li class="toctree-l2"><a class="reference internal" href="arguments.html">arguments module</a></li>
<li class="toctree-l2"><a class="reference internal" href="calling.html">calling module</a></li>
<li class="toctree-l2"><a class="reference internal" href="config.html">config module</a></li>
<li class="toctree-l2"><a class="reference internal" href="cropping.html">cropping module</a></li>
<li class="toctree-l2"><a class="reference internal" href="downloads.html">downloads module</a></li>
<li class="toctree-l2"><a class="reference internal" href="logger.html">logger module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline.html">pipeline module</a></li>
<li class="toctree-l2"><a class="reference internal" href="quality.html">quality module</a></li>
<li class="toctree-l2"><a class="reference internal" href="results.html">results module</a></li>
<li class="toctree-l2"><a class="reference internal" href="software.html">software module</a></li>
<li class="toctree-l2"><a class="reference internal" href="util.html">util module</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PLACEHOLDER</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">src</a></li>
      <li class="breadcrumb-item active">alignment module</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/alignment.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="module-alignment">
<span id="alignment-module"></span><h1>alignment module<a class="headerlink" href="#module-alignment" title="Permalink to this headline">¶</a></h1>
<p>This module handles everything about the alignment .</p>
<dl class="py function">
<dt id="alignment.extractUnaligned">
<code class="sig-prename descclassname">alignment.</code><code class="sig-name descname">extractUnaligned</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em><span class="sig-paren">)</span><a class="headerlink" href="#alignment.extractUnaligned" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts the unaligned sequences from the alignment against the 
host and discards the runs where all sequences aligned.</p>
<p>It filters the SAM files found in the 2-alignment/host/sam
directory with samtools view using the filter 256 and these
results are transformed into BAM files, found in the
2-alignment/host/bam directory. Then, using samtools bam2fq, it 
transforms the BAM files into FASTQ files with the final unaligned 
sequences.</p>
<p>In the case of paired-end runs, the resulting FASTQ has both 3’ and
5’ reads in the same file, so they are split into two files.</p>
<p>In the case of the unaligned FASTQ files generated by STAR, these 
are appended to the split FASTQ files.</p>
<p>The resulting FASTQ files are saved at 2-alignment/host/fastq.</p>
<dl class="simple">
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
</dd>
<dt>Returns:</dt><dd><dl class="simple">
<dt>list: List of filtered tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="alignment.runBWA">
<code class="sig-prename descclassname">alignment.</code><code class="sig-name descname">runBWA</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em>, <em class="sig-param"><span class="n">host</span></em><span class="sig-paren">)</span><a class="headerlink" href="#alignment.runBWA" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs BWA aligner.</p>
<p>It creates the index for each reference genome, then aligns each
run against each genome. Each run-genome pair generates a SAM file,
and it is saved in the 2-alignment/&lt;host/virus&gt;/sam directory.
It can take the arguments listed in the bwa.config file located
within the config directory.</p>
<dl>
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
<p>host (bool): Switch that tells if the alignment must be done against the host or the viral genomes.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="alignment.runGMAP">
<code class="sig-prename descclassname">alignment.</code><code class="sig-name descname">runGMAP</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em>, <em class="sig-param"><span class="n">host</span></em><span class="sig-paren">)</span><a class="headerlink" href="#alignment.runGMAP" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs GMAP aligner.</p>
<p>It creates the database for each reference genome, then aligns each
run against each genome. Each run-genome pair generates a SAM file,
and it is saved in the 2-alignment/&lt;host/virus&gt;/sam directory.
It can take the arguments listed in the gmap.config file located
within the config directory.</p>
<dl>
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
<p>host (bool): Switch that tells if the alignment must be done against the host or the viral genomes.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="alignment.runHisat2">
<code class="sig-prename descclassname">alignment.</code><code class="sig-name descname">runHisat2</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em>, <em class="sig-param"><span class="n">host</span></em><span class="sig-paren">)</span><a class="headerlink" href="#alignment.runHisat2" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs Hisat2 aligner.</p>
<p>It creates the database for each reference genome, then aligns each
run against each genome. Each run-genome pair generates a SAM file,
and it is saved in the 2-alignment/&lt;host/virus&gt;/sam directory.
It can take the arguments listed in the hisat2.config file located
within the config directory.</p>
<dl>
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
<p>host (bool): Switch that tells if the alignment must be done against the host or the viral genomes.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="alignment.runMagicBlast">
<code class="sig-prename descclassname">alignment.</code><code class="sig-name descname">runMagicBlast</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em>, <em class="sig-param"><span class="n">host</span></em><span class="sig-paren">)</span><a class="headerlink" href="#alignment.runMagicBlast" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs Magic-BLAST aligner.</p>
<p>It creates the database for each reference genome, then aligns each
run against each genome. Each run-genome pair generates a SAM file,
and it is saved in the 2-alignment/&lt;host/virus&gt;/sam directory.
It can take the arguments listed in the magicblast.config file 
located within the config directory.</p>
<dl>
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
<p>host (bool): Switch that tells if the alignment must be done against the host or the viral genomes.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="alignment.runMinimap2">
<code class="sig-prename descclassname">alignment.</code><code class="sig-name descname">runMinimap2</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em>, <em class="sig-param"><span class="n">host</span></em><span class="sig-paren">)</span><a class="headerlink" href="#alignment.runMinimap2" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs Minimap2 aligner.</p>
<p>It creates the database for each reference genome, then aligns each
run against each genome. Each run-genome pair generates a SAM file,
and it is saved in the 2-alignment/&lt;host/virus&gt;/sam directory.
It can take the arguments listed in the minimap2.config file 
located within the config directory.</p>
<dl>
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
<p>host (bool): Switch that tells if the alignment must be done against the host or the viral genomes.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="alignment.runSTAR">
<code class="sig-prename descclassname">alignment.</code><code class="sig-name descname">runSTAR</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em>, <em class="sig-param"><span class="n">host</span></em><span class="sig-paren">)</span><a class="headerlink" href="#alignment.runSTAR" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs STAR aligner.</p>
<p>It creates the database for each reference genome, then aligns each
run against each genome. Each run-genome pair generates a SAM file
with the sequences that may have a relevant alignment, which is
saved in the 2-alignment/&lt;host/virus&gt;/sam directory, and one or two 
FASTQ files (depending on the run type) with the unaligned 
sequences, which are saved in the 2-alignment/&lt;host/virus&gt;/star 
directory with .mate1 and .mate2 extensions. It can take the 
arguments listed in the star.config file located within the config 
directory.</p>
<dl>
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
<p>host (bool): Switch that tells if the alignment must be done against the host or the viral genomes.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="alignment.sortAlignments">
<code class="sig-prename descclassname">alignment.</code><code class="sig-name descname">sortAlignments</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sras</span></em><span class="sig-paren">)</span><a class="headerlink" href="#alignment.sortAlignments" title="Permalink to this definition">¶</a></dt>
<dd><p>Sorts the SAM files of each alignment against the viral genomes.</p>
<p>It sorts and transforms these SAM files into BAM files using the 
samtools sort tool, and then adds some extra headers needed for
following steps using samtools calmd. These files are saved in the 
2-alignment/virus/bam directory.</p>
<dl class="simple">
<dt>Args:</dt><dd><dl class="simple">
<dt>sras (list): List of tuples with the following data:</dt><dd><ul class="simple">
<li><p>A list of the paths for the input run (one file if single-end, two if paired-end)</p></li>
<li><p>Run type. “single” if single-end, “paired” if paired-end</p></li>
<li><p>Run ID</p></li>
</ul>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="modules.html" class="btn btn-neutral float-left" title="src" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="arguments.html" class="btn btn-neutral float-right" title="arguments module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, David Guevara.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>